src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1338,6 +1249,54 @@ internal static string GetModulePath()
             return currentModulePath;
         }
 
+#if !UNIX
+        /// <summary>
+        /// Returns a PSModulePath suiteable for Windows PowerShell by removing this PowerShell's specific
+        /// paths from current PSModulePath.
+        /// </summary>
+        /// <returns>
+        /// Returns appropriate PSModulePath for Windows PowerShell.
+        /// </returns>
+        internal static string GetWindowsPowerShellModulePath()
+        {
+            string currentModulePath = GetModulePath();
+
+            if (currentModulePath == null)
+            {
+                return null;
+            }
+
+            // Remove any PowerShell specific paths including if set in powershell.config.json file
```

Comment:
There is one scenario: pwsh 6/7 (pshome1) -> pwsh 7 (pshome2) -> powershell

In this case, when the second `pwsh` starts, the pshome1 module path will be kept in `PSModulePath`, then when starting powershell, the pshome2 module path will be scrubbed, but pshome1 module path will be kept and passed to powershell. Then powershell will load the built-in module from pshome1 path ...

Reply:
Correct, as discussed in the @PowerShell/powershell-committee meeting, this is not a case we want to optimize for since it's not likely and also PS6 support will end by July

Reply:
I'm fine that we don't support this, but I want to bring up that this will happen with pwsh7 or a preview version too, as long as the first pwsh is installed in a different location than the second pwsh.